src/tablespace.c

```diff
@@ -472,6 +493,18 @@ ts_tablespace_attach(PG_FUNCTION_ARGS)
 
 	ts_tablespace_attach_internal(tspcname, hypertable_oid, if_not_attached);
 
+	/* If the hypertable did not have a tablespace assigned, we set one */
+	rel = relation_open(hypertable_oid, AccessShareLock);
```

Comment:
I suggest to move this code part into separate function, eg: `attach_tablespace_to_hypertable` in the same way as been done for `detach_tablespace_from_hypertable_if_set()`. Then reuse this function for `detach_tablespace_from_hypertable_if_set` and other places to set `pg_default` tablespace instead of duplicating this functionality

Reply:
Hmmm... the conditions for running `AlterTableCmd` are different, so there is not much to gain by creating this function.